lib/trento/infrastructure/software_updates/auth/gen.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Trento.Infrastructure.SoftwareUpdates.Auth.Gen do
+  @moduledoc """
+  Behaviour of the SUMA authentication process.
+  """
+
+  alias Trento.Infrastructure.SoftwareUpdates.Suma.State
+
+  @callback authenticate() ::
+              {:ok, %State{}} | {:error, any()}
+
+  @callback authenticate(server_name :: String.t()) ::
```

Comment:
The tests definitely fail due this.
I will have a look, but I couldn't make an optional arg working in the callback

Reply:
sure tests need the argument, maybe the contract does not need it.

Reply:
But the mox mock is based on the contract though

Reply:
Yes, I went through same process and ended up in 
- not having the alternative signature in the contract
- still having the ability of mocking/testing things
- no complaints from credo/dialyzer/whatshoever

Everything was working without the need of the double signature.
Up to you.

Reply:
We can very well do that, but it would be a bit strange.
Default arguments are mere "macros" that help us create function signatures easier.
For example, in this case `authenticate/0` and `authenticate/1`. In essence, we have both.
Anyway, as we are just having this behaviour only to enable testing, I guess that it is fair to assume that the `server_name` is just a mere utility to make the tests possible, and not a real parameter that the user should pass